## SALES DATA ANALYSIS

### Initializing Project


In [39]:
#Importing Necessary Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [40]:
df = pd.read_csv('../Dataset/sales_data.csv')
df.head(15) 


ORDERNUMBER  QUANTITYORDERED  PRICEEACH  ORDERLINENUMBER    SALES  \
0         10107               30      95.70                2  2871.00   
1         10121               34      81.35                5  2765.90   
2         10134               41      94.74                2  3884.34   
3         10145               45      83.26                6  3746.70   
4         10159               49     100.00               14  5205.27   
5         10168               36      96.66                1  3479.76   
6         10180               29      86.13                9  2497.77   
7         10188               48     100.00                1  5512.32   
8         10201               22      98.57                2  2168.54   
9         10211               41     100.00               14  4708.44   
10        10223               37     100.00                1  3965.66   
11        10237               23     100.00                7  2333.12   
12        10251               28     100.00                2  3188.64   
13        10263               34     100.00                2  3676.76   
14        10275               45      92.83                1  4177.35   

          ORDERDATE   STATUS  QTR_ID  MONTH_ID  YEAR_ID  ...  \
0    2/24/2003 0:00  Shipped       1         2     2003  ...   
1   05/07/2003 0:00  Shipped       2         5     2003  ...   
2   07/01/2003 0:00  Shipped       3         7     2003  ...   
3    8/25/2003 0:00  Shipped       3         8     2003  ...   
4   10/10/2003 0:00  Shipped       4        10     2003  ...   
5   10/28/2003 0:00  Shipped       4        10     2003  ...   
6   11/11/2003 0:00  Shipped       4        11     2003  ...   
7   11/18/2003 0:00  Shipped       4        11     2003  ...   
8   12/01/2003 0:00  Shipped       4        12     2003  ...   
9    1/15/2004 0:00  Shipped       1         1     2004  ...   
10   2/20/2004 0:00  Shipped       1         2     2004  ...   
11  04/05/2004 0:00  Shipped       2         4     2004  ...   
12   5/18/2004 0:00  Shipped       2         5     2004  ...   
13   6/28/2004 0:00  Shipped       2         6     2004  ...   
14   7/23/2004 0:00  Shipped       3         7     2004  ...   

                     ADDRESSLINE1  ADDRESSLINE2           CITY     STATE  \
0         897 Long Airport Avenue           NaN            NYC        NY   
1              59 rue de l'Abbaye           NaN          Reims       NaN   
2   27 rue du Colonel Pierre Avia           NaN          Paris       NaN   
3              78934 Hillside Dr.           NaN       Pasadena        CA   
4                 7734 Strong St.           NaN  San Francisco        CA   
5               9408 Furth Circle           NaN     Burlingame        CA   
6         184, chausse de Tournai           NaN          Lille       NaN   
7     Drammen 121, PR 744 Sentrum           NaN         Bergen       NaN   
8       5557 North Pendale Street           NaN  San Francisco        CA   
9               25, rue Lauriston           NaN          Paris       NaN   
10              636 St Kilda Road       Level 3      Melbourne  Victoria   
11              2678 Kingston Rd.     Suite 101            NYC        NY   
12                  7476 Moss Rd.           NaN         Newark        NJ   
13            25593 South Bay Ln.           NaN    Bridgewater        CT   
14   67, rue des Cinquante Otages           NaN         Nantes       NaN   

   POSTALCODE    COUNTRY TERRITORY CONTACTLASTNAME CONTACTFIRSTNAME DEALSIZE  
0       10022        USA       NaN              Yu             Kwai    Small  
1       51100     France      EMEA         Henriot             Paul    Small  
2       75508     France      EMEA        Da Cunha           Daniel   Medium  
3       90003        USA       NaN           Young            Julie   Medium  
4         NaN        USA       NaN           Brown            Julie   Medium  
5       94217        USA       NaN          Hirano             Juri   Medium  
6       59000     France      EMEA      

In [41]:
# SHAPE AND COLUMN NAMES OF DATASET
df.shape, df.columns

((2823, 25),
 Index(['ORDERNUMBER', 'QUANTITYORDERED', 'PRICEEACH', 'ORDERLINENUMBER',
        'SALES', 'ORDERDATE', 'STATUS', 'QTR_ID', 'MONTH_ID', 'YEAR_ID',
        'PRODUCTLINE', 'MSRP', 'PRODUCTCODE', 'CUSTOMERNAME', 'PHONE',
        'ADDRESSLINE1', 'ADDRESSLINE2', 'CITY', 'STATE', 'POSTALCODE',
        'COUNTRY', 'TERRITORY', 'CONTACTLASTNAME', 'CONTACTFIRSTNAME',
        'DEALSIZE'],
       dtype='object'))

### DATA INSPECTION


In [50]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2823 entries, 0 to 2822
Data columns (total 25 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   ORDERNUMBER       2823 non-null   int64  
 1   QUANTITYORDERED   2823 non-null   int64  
 2   PRICEEACH         2823 non-null   float64
 3   ORDERLINENUMBER   2823 non-null   int64  
 4   SALES             2823 non-null   float64
 5   ORDERDATE         2823 non-null   object 
 6   STATUS            2823 non-null   object 
 7   QTR_ID            2823 non-null   int64  
 8   MONTH_ID          2823 non-null   int64  
 9   YEAR_ID           2823 non-null   int64  
 10  PRODUCTLINE       2823 non-null   object 
 11  MSRP              2823 non-null   int64  
 12  PRODUCTCODE       2823 non-null   object 
 13  CUSTOMERNAME      2823 non-null   object 
 14  PHONE             2823 non-null   object 
 15  ADDRESSLINE1      2823 non-null   object 
 16  ADDRESSLINE2      302 non-null    object 


In [48]:
df.isna().sum()

ORDERNUMBER            0
QUANTITYORDERED        0
PRICEEACH              0
ORDERLINENUMBER        0
SALES                  0
ORDERDATE              0
STATUS                 0
QTR_ID                 0
MONTH_ID               0
YEAR_ID                0
PRODUCTLINE            0
MSRP                   0
PRODUCTCODE            0
CUSTOMERNAME           0
PHONE                  0
ADDRESSLINE1           0
ADDRESSLINE2        2521
CITY                   0
STATE               1486
POSTALCODE            76
COUNTRY                0
TERRITORY           1074
CONTACTLASTNAME        0
CONTACTFIRSTNAME       0
DEALSIZE               0
dtype: int64

### OBSERVATION
- Number of Rows and Column: `2823, 25`
- Column-Wise Missing Values: [ADDRESSLINE2 : `2521`,
                    STATE: `1486`,
                    POSTAL CODE: `76`,
                    TERRITORY: `1074`]
- dtypes: `float64(2), int64(7), object(16)`